modules/assim.sequential/inst/covariates.R

```diff
@@ -0,0 +1,109 @@
+# Product is a zip file containing 12 .tif files, one for each month
+# Use code below to average the 12 files to obtain one map
+# 2023 REU project used 10 minute spatial resolution
+
+## Solar Radiation (kJ m-2 day-1)
+srad <- terra::rast(list.files(path = "/projectnb/dietzelab/jploshay/pecan_copy/jploshay/10m_srad",
+                               pattern='.tif$',
+                               all.files= T,
+                               full.names= T))
+srad <- terra::app(srad, mean)
+
+
+## Vapor Pressure (kPa)
+vapr <- terra::rast(list.files(path = "/projectnb/dietzelab/jploshay/pecan_copy/jploshay/10m_vapr",
+                               pattern='.tif$',
+                               all.files= T,
+                               full.names= T)) 
+vapr <- terra::app(vapr, mean)
+
+
+## Average Temperature (*C)
+tavg <- terra::rast(list.files(path = "/projectnb/dietzelab/jploshay/pecan_copy/jploshay/avg_temp_prep/WorldClim",
+                               pattern='.tif$',
+                               all.files= T,
+                               full.names= T))
+tavg <- terra::app(tavg, mean)
+
+
+## Total Precipitation (mm)
+prec <- terra::rast(list.files(path = "/projectnb/dietzelab/jploshay/pecan_copy/jploshay/total_prec",
+                               pattern='.tif$',
+                               all.files= T,
+                               full.names= T))
+prec <- terra::app(prec, mean)
+
+
+#### SoilGrids ####
+# geodata::soil_world() pulls data from the SoilGRIDS database
+# More information on pulling different soil data can be found in the geodata 
+# manual https://cran.r-project.org/web/packages/geodata/geodata.pdf
+
+## Soil pH
+phh2o <- geodata::soil_world(var = "phh2o", depth = 5, stat = "mean", path = tempdir()) 
+
+## Soil Nitrogen (g kg-1)
+nitrogen <- geodata::soil_world(var = "nitrogen", depth = 5, stat = "mean", path = tempdir())
+
+## Soil Organic Carbon (g kg-1)
+soc <- geodata::soil_world(var = "soc", depth = 5, stat = "mean", path = tempdir())
+
+## Soil Sand (%)
+sand <- geodata::soil_world(var = "sand", depth = 5, stat = "mean", path = tempdir())
+
+
+#### Land Cover ####
+GLanCE_extract <- function(pattern, path) {
+  files <- list.files(path = "/projectnb/dietzelab/dietze/glance2012/e4ftl01.cr.usgs.gov/MEASURES/GLanCE30.001/2012.07.01", #make this path default
+                      all.files = T,
+                      full.names = T,
+                      pattern)
+  # empty .vrt file path
+  vrtfile <- paste0(tempfile(), ".vrt") 
+  # "connects" tiles together that returns SpatRaster
+  GLanCE_product <- terra::vrt(files, vrtfile, overwrite = T) 
+  return(GLanCE_product)
+}
+
+# Integer identifier for class in the current year
+land_cover <- GLanCE_extract(pattern = "NA_LC.tif$")
+
+
+#### Resample and Stack Maps ####
+# Define the extent to crop the covariates to North America
+NA_extent <- terra::ext(-178.19453125, -10, 7.22006835937502, 83.5996093750001)
```

Comment:
This would be good to have at the top of the script in case a user wants/needs to set a different extent.